// @unovis/ts/components/treemap/style.js
import { css } from '@emotion/css';
import { getCssVarNames, injectGlobalCssVariables } from '../../utils/style.js';

const cssVarDefaults = {
    '--vis-treemap-tile-stroke-color': '#fff',
    '--vis-treemap-tile-stroke-width': '2px',
    '--vis-treemap-tile-hover-stroke-color': '#fff',
    '--vis-treemap-tile-hover-stroke-opacity': 0,
    '--vis-treemap-tile-fill-color': '#B9BEC3',
    '--vis-treemap-tile-background-color': '#fff',
    '--vis-treemap-tile-cursor': 'default',
    '--vis-treemap-label-text-color': '#000',
    '--vis-treemap-label-font-size': '12px',
    /* Label opacity */
    '--vis-treemap-label-opacity': 0.8,
    '--vis-treemap-label-font-weight': 'normal',
    /* Dark Theme */
    '--vis-dark-treemap-tile-stroke-color': '#2c2c2c',
    '--vis-dark-treemap-tile-fill-color': '#5b5f6d',
    '--vis-dark-treemap-label-text-color': '#5b5f6d',
};
const root = css `
  label: treemap-component;
`;
const variables = getCssVarNames(cssVarDefaults);
injectGlobalCssVariables(cssVarDefaults, root);
const tiles = css `
  label: g-tiles;
`;
const tileGroup = css `
  label: tile-group;
`;
const tile = css `
  label: tile;
  stroke: var(${variables.treemapTileHoverStrokeColor});
  stroke-opacity: 0;

  &:hover {
    stroke-opacity: var(--vis-treemap-tile-hover-stroke-opacity);
  }
`;
// The leaf tiles are clickable
const clickableTile = css `
  label: clickable-tile;
  cursor: pointer;
`;
const tileForeground = css `
  label: tile-foreground;
`;
const label = css `
  label: label;
  text-anchor: start;
  dominant-baseline: hanging;
  user-select: none;
  pointer-events: none;
  font-size: var(--vis-treemap-label-font-size);
  opacity: var(--vis-treemap-label-opacity);
  fill: var(--vis-treemap-label-text-color);
  font-weight: var(--vis-treemap-label-font-weight);
`;
const internalLabel = css `
  font-weight: 500;
`;
const labelGroup = css `
  label: label-group;
`;

export { clickableTile, internalLabel, label, labelGroup, root, tile, tileForeground, tileGroup, tiles, variables };
